### PR TITLE
eth_nxp_enet_qos: implement set_config

### PR DESCRIPTION
**Pull request by theadib**
implements  the set_config api for the eth_nxp_enet_qos_mac driver 
as in FRDM_MCXN947

### CODE REVIEW

**Comment by decsny** [comment]
Hi, the PR looks good at a glance, I will review more when I get a chance. For now make sure you ran the scripts/ci/check_compliance script and passed (expect clang format check is optional, the driver might already not pass clang format)

**Comment by dleach02** [comment]
@theadib thankyou for the contributions. Can you address the compliance issue:

 Commit message body line exceeds max length (85>75): "implements  the set_config api for the eth_nxp_enet_qos_mac driver as in FRDM_MCXN947"

**Review by decsny — APPROVED** [review]
LGTM

File: drivers/ethernet/eth_nxp_enet_qos/eth_nxp_enet_qos_mac.c [hunk 1]
```diff
@@ -570,11 +570,55 @@ static const struct device *eth_nxp_enet_qos_get_phy(const struct device *dev)
 	return config->phy_dev;
 }
 
+
+
+static int eth_nxp_enet_qos_set_config(const struct device *dev,
+			       enum ethernet_config_type type,
+			       const struct ethernet_config *cfg)
```
Comment (pdgendt) [inline drivers/ethernet/eth_nxp_enet_qos/eth_nxp_enet_qos_mac.c:577]: Nit: alignment

```suggestion
static int eth_nxp_enet_qos_set_config(const struct device *dev,
				       enum ethernet_config_type type,
				       const struct ethernet_config *cfg)
```

File: drivers/ethernet/eth_nxp_enet_qos/eth_nxp_enet_qos_mac.c [hunk 2]
```diff
@@ -570,11 +570,55 @@ static const struct device *eth_nxp_enet_qos_get_phy(const struct device *dev)
...
+
+	return -ENOTSUP;
+}
+
 static const struct ethernet_api api_funcs = {
 	.iface_api.init = eth_nxp_enet_qos_iface_init,
 	.send = eth_nxp_enet_qos_tx,
 	.get_capabilities = eth_nxp_enet_qos_get_capabilities,
 	.get_phy = eth_nxp_enet_qos_get_phy,
+	.set_config	= eth_nxp_enet_qos_set_config,
```
Comment (pdgendt) [inline drivers/ethernet/eth_nxp_enet_qos/eth_nxp_enet_qos_mac.c:621]: Nit space vs tab

```suggestion
	.set_config = eth_nxp_enet_qos_set_config,
```

**Review by pdgendt — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Pull request merged**